<launch>
    <!-- Detection configuration -->
    <arg name="weights" default="yolov8n.pt" />
    <arg name="confidence_threshold" default="0.75" />
    <arg name="iou_threshold" default="0.45" />
    <arg name="maximum_detections" default="300" />
    <arg name="agnostic_nms" default="true" />

    <!-- ROS topics -->
    <arg name="input_image_topic" default="/camera/color/image_raw/compressed" />
    <arg name="output_topic" default="/yolo_detector/detections" />

    <!-- Optional topic (publishing annotated image) -->
    <arg name="publish_image" default="false" />
    <arg name="output_image_topic" default="/yolo_detector/image_out" />

    <!-- Output verbose info -->
    <arg name="verbose" default="false" />

    <node pkg="yolo_detector_ros" name="detect" type="detect.py" output="screen">
        <param name="weights" value="$(arg weights)" />
        <param name="confidence_threshold" value="$(arg confidence_threshold)" />
        <param name="iou_threshold" value="$(arg iou_threshold)" />
        <param name="maximum_detections" value="$(arg maximum_detections)" />
        <param name="agnostic_nms" value="$(arg agnostic_nms)" />        
        <param name="input_image_topic" value="$(arg input_image_topic)" />
        <param name="output_topic" value="$(arg output_topic)" />
        <param name="publish_image" value="$(arg publish_image)" />
        <param name="output_image_topic" value="$(arg output_image_topic)" />
        <param name="verbose" value="$(arg verbose)" />
    </node>

</launch>